oauthenticator/generic.py

```diff
@@ -170,6 +192,25 @@ class GenericOAuthenticator(OAuthenticator):
             }
         }
 
+    async def _check_group_whitelist(self, http_client, username, headers):
+        if self.groupdata_url:
+            url = self.groupdata_url
+        else:
+            raise ValueError("Please set the OAUTH2_GROUPDATA_URL environment variable")
+
+        req = HTTPRequest(url,
+                          method=self.groupdata_method,
+                          headers=headers,
+                          validate_cert=self.tls_verify,
+                          )
+        resp = await http_client.fetch(req)
+        resp_json = json.loads(resp.body.decode('utf8', 'replace'))
+        user_groups = resp_json.get('groups')
```

Comment:
Is the 'groups' field and structure generic to oauth? What provider have you used this with?

Reply:
I think this is really provider specific. Even fetching data from group_data_url is provider specific. Usually group member ship is either in an id_token (OIDC), can be fetched from a user info endpoint or maybe taken from an access token (or token introspection endpoint). Anything else is outside of the oauth2 or OIDC spec.